src/coreComponents/mesh/generators/WellGeneratorBase.cpp

```diff
@@ -528,54 +529,49 @@ void WellGeneratorBase::debugWellGeometry() const
     return;
   }
 
-  std::cout << std::endl;
-  std::cout << "++++++++++++++++++++++++++" << std::endl;
-  std::cout << "WellGeneratorBase = " << getName() << std::endl;
-  std::cout << "MPI rank = " << MpiWrapper::commRank( MPI_COMM_GEOSX ) << std::endl << std::endl;
-  std::cout << "Number of well elements = " << m_numElems << std::endl;
+  std::vector< string > row;
+  std::ostringstream oss;
+  Table table = Table( {
```

Comment:
Maybe, instead of using a `Table` directly here, you could have stored the data in an appropriate container (maybe something around the `HistoryCollection` abstraction, or a simple `std::map`)?
Then we can select the dump format as a separate step? That would be more flexible.

Reply:
I don't see how a historyCollection could produce a clear code.
I think it's easier to use a struct to store data and to use a condition in order to display the table or the legacy log

```c++
struct WellElementLog
  {
    arraySlice1d< real64 > coordinates;
    std::optional< globalIndex > nextElement;
    std::optional< globalIndex > prevElement;
  };

  std::map< int, WellElementLog > elements;
  for( globalIndex iwelem = 0; iwelem < m_numElems; ++iwelem )
  {
    WellElementLog elem;
    if( m_nextElemId[iwelem] >= 0 )
    {
      elem.nextElement =  m_nextElemId[iwelem];
    }
    if( m_prevElemId[iwelem][0] >= 0 )
    {
      elem.prevElement =  m_prevElemId[iwelem][0];
    }
    elem = m_elemCenterCoords[iwelem];
    elements[iwelem] = elem;
  }

  // if ( isUsingTableFormat ... )
  Table table = Table( {
      Table::ColumnParam{{"Element no."}, Table::Alignment::right},
      Table::ColumnParam{{"CoordX"}, Table::Alignment::middle},
      Table::ColumnParam{{"CoordY"}, Table::Alignment::middle},
      Table::ColumnParam{{"CoordZ"}, Table::Alignment::middle},
      Table::ColumnParam{{"Prev\nElement"}, Table::Alignment::right},
      Table::ColumnParam{{"Next\nElement"}, Table::Alignment::right},
    } );
  string titleName = "InternalWellGenerator " + getName();
  table.setTitle( titleName );
  for( const auto & [iwelem, elem] : elements )
  {
    table.addRow( iwelem, elem.coordinates[0], elem.coordinates[1], elem.coordinates[2],
                  elem.nextElement, elem.prevElement )
  }
  table.draw();
 ```

Reply:
Please at least separate the collection and storage of the raw data from its display.
E.g. you could have a `Table` that displays on `std::cout` and another one that stores the data as a `json` file easy to parse. The two based on the on the same data input.

The current `TimeHistory` pattern separated, IIRC, the collection, the storage and the dump. Maybe could you have a look at how it's done?

Reply:
Do you think the code I posted earlier  does separate the collection and the dump ?
Does the fact that the table stores the parsed data lead to an issue ?